packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -7979,16 +7981,20 @@ export default class Meeting extends StatelessWebexPlugin {
     }
 
     const permissionTokenExpValue = Number(this.permissionTokenPayload.exp);
+    const permissionTokenIssuedTime = Number(this.permissionTokenPayload.iat);
+
+    const shiftInTime = this.tokenReceivedTime - permissionTokenIssuedTime;
 
     // using new Date instead of Date.now() to allow for accurate unit testing
     // https://github.com/sinonjs/fake-timers/issues/321
-    const now = new Date().getTime();
+    const currentTime = new Date().getTime();
+
+    // adjusted time is calculated in case your machine time is wrong
+    const adjustedCurrentTime = currentTime - shiftInTime;
 
-    // substract current time from the permissionTokenExp
-    // (permissionTokenExp is a epoch timestamp, not a time to live duration)
-    const timeLeft = (permissionTokenExpValue - now) / 1000;
+    const timeLeft = (permissionTokenExpValue - adjustedCurrentTime) / 1000;
 
-    return {timeLeft, expiryTime: permissionTokenExpValue, currentTime: now};
+    return {timeLeft, expiryTime: permissionTokenExpValue, currentTime};
```

Comment:
issue: if we adjust the `timeLeft` then we should do the same for `expiryTime`, so `expiryTime` should be `permissionTokenExpValue - shiftInTime`

Reply:
No, expiryTime is being received in permissionToken payload, we are not adjusting timeLeft we are adjusting the current local time